examples/hybrid/staggered_nonhydrostatic_model.jl

```diff
@@ -766,12 +736,49 @@ function Wfact_generic!(W, Y, p, dtγ, t)
         # ᶠw == ᶠw_data .* ᶠw_unit. The Jacobian blocks involve ᶠw_data, not ᶠw.
         ᶠw_data = ᶠw.components.data.:1
 
-        # If ∂(ᶜarg)/∂(ᶠw_data) = 0, then
-        # ∂(ᶠupwind_product(ᶠw, ᶜarg))/∂(ᶠw_data) =
-        #     ᶠupwind_product(ᶠw + εw, arg) / to_scalar(ᶠw + εw).
+        # To convert ᶠw to ᶠw_data, we extract the third vector component.
+        to_scalar(vector) = vector.u₃
+
+        # To convert ∂(ᶠwₜ)/∂(ᶜ𝔼) to ∂(ᶠw_data)ₜ/∂(ᶜ𝔼) and ∂(ᶠwₜ)/∂(ᶠw_data) to
+        # ∂(ᶠw_data)ₜ/∂(ᶠw_data), we extract the third component of each vector-
+        # valued stencil coefficient.
+        to_scalar_coefs(vector_coefs) =
+            map(vector_coef -> vector_coef.u₃, vector_coefs)
+
+        # If ᶜρcₜ = -ᶜdivᵥ(ᶠinterp(ᶜρc) * ᶠw), then
+        # ∂(ᶜρcₜ)/∂(ᶠw_data) =
+        #     -ᶜdivᵥ_stencil(ᶠinterp(ᶜρc) * ᶠw_unit) -
+        #     ᶜdivᵥ_stencil(ᶠw) * ᶠinterp_stencil(1) * ∂(ᶜρc)/∂(ᶠw_data)
+        # If ᶜρcₜ = -ᶜdivᵥ(ᶠinterp(ᶜρ) * ᶠupwind(ᶠw, ᶜρc / ᶜρ)), then
+        # ∂(ᶜρcₜ)/∂(ᶠw_data) =
+        #     -ᶜdivᵥ_stencil(ᶠinterp(ᶜρc) *
+        #     ᶠupwind(ᶠw + εw, ᶜρc) / to_scalar(ᶠw + εw)) -
+        #     ᶜdivᵥ_stencil(ᶠinterp(ᶜρ)) * ᶠupwind_stencil(ᶠw, 1 / ᶜρ) *
+        #     ∂(ᶜρc)/∂(ᶠw_data)
```

Comment:
All of these (and other mathy comments) should really be part of our docs. Not all browsers or IDEs render math/unicode symbols the same, but we have a way of rendering them in a cross-platform fashion via Documenter.jl :)

Reply:
These comments are purely for my own derivations and sanity-checking. They are not intended to be documentation for ClimaAtmos. I will eventually write some documentation for how to extend the implicit solver, but this will only come after the overall interface update, which I expect will drastically change how the implicit solver works.